server/src/main/java/org/opensearch/indices/replication/RemoteStoreReplicationSource.java

```diff
@@ -93,15 +96,43 @@ public void getSegmentFiles(
         ActionListener<GetSegmentFilesResponse> listener
     ) {
         try {
-            indexShard.syncSegmentsFromRemoteSegmentStore(false, true, false);
-            listener.onResponse(new GetSegmentFilesResponse(Collections.emptyList()));
+            logger.trace("Downloading segments files from remote store {}", filesToFetch);
+            FilterDirectory remoteStoreDirectory = (FilterDirectory) indexShard.remoteStore().directory();
+            FilterDirectory byteSizeCachingStoreDirectory = (FilterDirectory) remoteStoreDirectory.getDelegate();
+            RemoteSegmentStoreDirectory remoteSegmentStoreDirectory = (RemoteSegmentStoreDirectory) byteSizeCachingStoreDirectory
+                .getDelegate();
+            RemoteSegmentMetadata remoteSegmentMetadata = remoteSegmentStoreDirectory.init();
+            List<StoreFileMetadata> downloadedSegments = new ArrayList<>();
+            if (remoteSegmentMetadata != null) {
+                try {
+                    indexShard.store().incRef();
+                    indexShard.remoteStore().incRef();
+                    final Directory storeDirectory = indexShard.store().directory();
+                    String segmentNFile = null;
+                    for (StoreFileMetadata fileMetadata : filesToFetch) {
+                        String file = fileMetadata.name();
+                        storeDirectory.copyFrom(remoteSegmentStoreDirectory, file, file, IOContext.DEFAULT);
+                        downloadedSegments.add(fileMetadata);
+                        if (file.startsWith(IndexFileNames.SEGMENTS)) {
```

Comment:
With this change, we'd now skip `segment_N` files while downloading for SegRep with remote store as well. 
Should we have a common logic to exclude `segment_N` in the `SegmentReplicationTarget.getFiles` instead?

Reply:
There is no change in the logic around downloading files during round of segment replication. This check validates that there is only one `segment_N` file.